core/src/components/cat-radio-group/cat-radio-group.tsx

```diff
@@ -0,0 +1,76 @@
+import { Component, Host, h, Prop, Element, Listen } from '@stencil/core';
+
+@Component({
+  tag: 'cat-radio-group',
+  styleUrl: 'cat-radio-group.scss',
+  shadow: true
+})
+export class CatRadioGroup {
+  private catRadioGroup: HTMLCatRadioElement[] = [];
+
+  @Element() hostElement!: HTMLElement;
+
+  /**
+   * The name of the radio group component.
+   */
+  @Prop() name?: string;
+
+  /**
+   * Whether this radio group is disabled.
+   */
+  @Prop() disabled = false;
+
+  componentDidLoad(): void {
+    this.catRadioGroup = Array.from(this.hostElement.querySelectorAll(`cat-radio`));
+    this.catRadioGroup.forEach(catRadio => {
+      if (this.disabled) {
+        catRadio.disabled = this.disabled;
+      }
+      catRadio.name = this.name;
+    });
+    this.updateTabIndex();
+  }
+
+  @Listen('keydown')
+  onKeydown(event: KeyboardEvent): void {
+    if (['ArrowDown', 'ArrowUp', 'ArrowRight', 'ArrowLeft'].includes(event.key) && this.catRadioGroup.length) {
+      const targetElements = this.catRadioGroup.filter(catRadio => !catRadio.disabled);
+      const activeElement = document.activeElement as HTMLCatRadioElement;
+      const activeIdx = this.catRadioGroup.findIndex(catRadio => catRadio === activeElement);
+      const activeOff = ['ArrowDown', 'ArrowRight'].includes(event.key) ? 1 : -1;
+      const targetIdx = activeIdx < 0 ? 0 : (activeIdx + activeOff + targetElements.length) % targetElements.length;
+      targetElements[targetIdx].setFocus();
+      targetElements[targetIdx].shadowRoot?.querySelector('input')?.click();
+      this.updateTabIndex();
+      event.preventDefault();
+    }
+  }
+
+  @Listen('click')
+  onClick(event: MouseEvent): void {
```

Comment:
do we need to check if it is disabled and then do an early return?

Reply:
By changing it to onInput, it would no longer be necessary